sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs

```diff
@@ -385,13 +398,34 @@ public static void SendEvent_TestHub(string input, [EventHub(TestHubName, Connec
 
             public static void ProcessSingleEvent([EventHubTrigger(TestHubName, Connection = TestHubName)] string evt,
                 string partitionKey, DateTime enqueuedTimeUtc, IDictionary<string, object> properties,
-                IDictionary<string, object> systemProperties)
+                IDictionary<string, object> systemProperties,
+                PartitionContext partitionContext)
             {
                 Assert.True((DateTime.Now - enqueuedTimeUtc).TotalSeconds < 30);
 
                 Assert.AreEqual("value1", properties["TestProp1"]);
                 Assert.AreEqual("value2", properties["TestProp2"]);
 
+                Assert.NotNull(partitionContext.PartitionId);
+                Assert.NotNull(partitionContext.ReadLastEnqueuedEventProperties());
+
+                _eventWait.Set();
+            }
+        }
+
+        public class EventHubTestClientDispatch
+        {
+            public static async Task SendEvents([EventHub(TestHubName, Connection = TestHubName)] EventHubProducerClient producer)
+            {
+                await producer.SendAsync(new[]
+                {
+                    new EventData(new BinaryData("Event 1")),
+                });
+            }
+
+            public static void ProcessSingleEvent([EventHubTrigger(TestHubName, Connection = TestHubName)] EventData eventData)
```

Comment:
Shouldn't there be an end to end test that binds to PartitionContext?

Reply:
I added one in this PR https://github.com/Azure/azure-sdk-for-net/pull/19114/files#diff-ebf190a222702ad012429237484e3b2cc982fec69efad485d34808b90ecd6121R402